core/src/main/scala/org.broadinstitute.dsde.workbench.leonardo/models.scala

```diff
@@ -1,19 +1,146 @@
 package org.broadinstitute.dsde.workbench.leonardo
 
+import java.net.URL
 import java.time.Instant
 
+import cats.implicits._
 import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, parseGcsPath}
+import enumeratum.{Enum, EnumEntry}
+import ca.mrvisser.sealerate
 
-final case class MachineConfig(numberOfWorkers: Option[Int] = None,
-                               masterMachineType: Option[String] = None,
-                               masterDiskSize: Option[Int] = None, //min 10
-                               workerMachineType: Option[String] = None,
-                               workerDiskSize: Option[Int] = None, //min 10
-                               numberOfWorkerLocalSSDs: Option[Int] = None, //min 0 max 8
-                               numberOfPreemptibleWorkers: Option[Int] = None)
+import scala.collection.immutable
+import scala.util.matching.Regex
+
+
+final case class MachineType(value: String) extends AnyVal
+
+sealed trait CloudService extends EnumEntry with Product with Serializable {
+  def asString: String
+
+  override def toString: String = asString //Enumeratum's withName function uses `toString` as key for lookup
+}
+object CloudService extends Enum[CloudService]{
+  case object Dataproc extends CloudService {
+    val asString: String = "DATAPROC"
+  }
+  case object GCE extends CloudService {
+    val asString: String = "GCE"
```

Comment:
Thinking out loud… should we name things `GCE` or `Compute`?

Reply:
`Compute` sounds generic to me like `CloudService` so I'd vote for `GCE`.

Reply:
yea. I prefer gce as well

Reply:
ok